packages/xo-web/src/xo-app/dashboard/overview/index.js

```diff
@@ -68,29 +70,82 @@ class PatchesCard extends Component {
   }
 }
 
-@connectStore(() => {
-  const getHosts = createGetObjectsOfType('host')
-  const getVms = createGetObjectsOfType('VM')
-
-  const getHostMetrics = createGetHostMetrics(getHosts)
+@connectStore({
+  hosts: createGetObjectsOfType('host'),
+  pools: createGetObjectsOfType('pool'),
+  srs: createGetObjectsOfType('SR').filter([isSrWritable]),
+  vms: createGetObjectsOfType('VM'),
+  alarmMessages: createGetObjectsOfType('message').filter([
+    message => message.name === 'ALARM',
+  ]),
+  tasks: createGetObjectsOfType('task').filter([
+    task => task.status === 'pending',
+  ]),
+})
+@addSubscriptions({
+  plugins: subscribePlugins,
+  users: subscribeUsers,
+})
+@injectIntl
+class DefaultCard extends Component {
+  _getPoolWisePredicate = createSelector(
+    () => map(this.state.pools, 'id'),
+    poolsIds => item => isEmpty(poolsIds) || includes(poolsIds, item.$pool)
+  )
 
-  const writableSrs = createGetObjectsOfType('SR').filter([isSrWritable])
+  _getPredicate = createSelector(
+    this._getPoolWisePredicate,
+    () => map(this.state.hosts, 'id'),
+    (poolWisePredicate, hostsIds) => item =>
+      isEmpty(hostsIds)
+        ? poolWisePredicate(item)
+        : includes(hostsIds, item.$container || item.$host)
+  )
 
-  const getSrMetrics = createCollectionWrapper(
-    createSelector(writableSrs, writableSrs => {
-      const metrics = {
-        srTotal: 0,
-        srUsage: 0,
-      }
-      forEach(writableSrs, sr => {
-        metrics.srUsage += sr.physical_usage
-        metrics.srTotal += sr.size
-      })
-      return metrics
+  _onPoolsChange = pools => {
+    const { hosts } = this.state
+    this.setState({
+      pools,
+      hosts: isEmpty(pools)
+        ? hosts
+        : filter(hosts, host => includes(map(pools, 'id'), host.$pool)),
```

Comment:
Avoid building the array again everytime: `const poolIds = map(pools, 'id')` before the `setState`.